docs/open_source/release-process.md

```diff
@@ -18,80 +23,215 @@ This will log you into NPM.
-> remote: error: GH006: Protected branch update failed for refs/heads/master.
-> remote: error: Required status check "cla/google" is expected. At least one approved review is required by reviewers with write access.
-> To github.com:material-components/material-components-web.git
-> ! [remote rejected]   master -> master (protected branch hook declined)
-> ```
-> You may need to update Github's master branch protection:
-> 1. Go to: [settings page](https://github.com/material-components/material-components-web/settings/branches/master)
-> 1. Uncheck `Include administrators`
-> 1. Click Save changes
-> 1. Perform `git push && git push --tags`
-> 1. Don't forget to toggle on `Include administrators` & click Save changes
+```
+npm run changelog
+git diff # Review the changelog and make sure it looks OK
+./scripts/post-release.sh
+```
+
+### For Minor Releases
+
+> Note: In the rare case there were no pre-releases leading up to this release, you can follow the same steps above.
+
+First, update the changelog without committing it:
+
+```
+npm run changelog
+```
+
+Next, edit the changelog:
+
+* Bring any changes from the prior pre-releases under their respective headings for the new final release
+* Remove headings for the pre-releases
+
+See [this v0.36.0 commit](https://github.com/material-components/material-components-web/commit/13fd6784866864839d0d287b3703b3beb0888d9c)
+for an example of the resulting changes.
+
+This will make the changelog easier to read, since users won't be interested in the pre-releases once the final is
+tagged, and shouldn't need to read the new release's changes across multiple headings.
+
+Finally, run the post-release script to commit the updated changelog and apply a git tag:
+
+```
+./scripts/post-release.sh
+```
+
+## Push
+
+### Common First Step for All Releases
+
+You will need to temporarily alter Github's master branch protection in order to push after the release:
+
+1. Go to: [settings page](https://github.com/material-components/material-components-web/settings/branches/master)
+1. Uncheck `Include administrators`
+1. Click Save changes
+1. Perform the process outlined in one of the sections below
+1. Don't forget to toggle on `Include administrators` & click Save changes afterwards
+
+### For Pre-releases and Feature/Breaking-Change Releases
+
+`git push origin master && git push origin <tag>`
+
+This will ensure the new commits *and* tag are pushed to the remote git repository.
+
+### For Bugfix Releases
+
+`git push origin <tag>`
+
+We don't need to push a branch for bugfix releases since we only cherry-pick commits for them at release time and they
+are not tagged from master (which contains all commits, not just bugfixes). However, we *do* need to sync the new
+release versions and changelog with master.
+
+Run `git log` and take note of the publish and changelog commit hashes. Then switch to master and cherry-pick them:
+
+```
+git checkout master
+git cherry-pick -x <publish-hash> <changelog-hash>
```

Comment:
Is this cherry-picking both 2 commits at the same time?

Reply:
Yes; they still remain as 2 separate commits - it's the same as running the cherry-pick command twice with each commit separately.